packages/sitecore-jss/src/graphql-request-client.test.ts

```diff
@@ -330,6 +387,76 @@ describe('GraphQLRequestClient', () => {
         });
       }
 
+      const retryableErrorCodeThrowError = async (errorCode: string) => {
+        nock('http://jssnextweb')
+          .post('/graphql')
+          .replyWithError({ code: errorCode })
+          .post('/graphql')
+          .replyWithError({ code: errorCode })
+          .post('/graphql')
+          .replyWithError({ code: errorCode });
+
+        const graphQLClient = new GraphQLRequestClient(endpoint, {
+          retries: 2,
+        });
+
+        spy.on(graphQLClient['client'], 'request');
+
+        try {
+          await graphQLClient.request('test');
+        } catch (error) {
+          expect(error).to.not.be.undefined;
+          expect(graphQLClient['client'].request).to.have.been.called.exactly(3);
+          spy.restore(graphQLClient);
+        }
+      };
+
+      // Test cases for each retryable error codes
+      for (const errorCode of ['ECONNRESET', 'ETIMEDOUT', 'EPROTO']) {
```

Comment:
I don't understand this one exactly. What is duplicated here? Because we cannot combine both error types.
Or you just want to test some of the error codes and not all?

Reply:
If I am understanding this correctly then we cannot reuse the array and combine both the types of error codes into one like the following since they both use different method for reply in the `nock` request.
`
const nodeErrorCode = ['ECONNRESET', 'ETIMEDOUT', 'EPROTO']
const statusCode = [429, 502, 503, 504, 520, 521, 522, 523, 524];
for (const errorCode of [...statusCode, ...nodeErrorCode] ){
    ...
}
`

Reply:
Here is what you need to update
1. Copy array from [line](https://github.com/Sitecore/jss/pull/1755/files#diff-50af470e1125379203c7b1f723dbc4c9d15aba853dc2f2ca351ad52067b10177R478)
2. Define it as a separate variable
3. Use this variable in a step 1 and in a [line](https://github.com/Sitecore/jss/pull/1755/files#diff-50af470e1125379203c7b1f723dbc4c9d15aba853dc2f2ca351ad52067b10177R440)

Reply:
ohhh got it now
I totally misunderstood this comment previously.